src/filters/renderer/VideoRenderers/DX9RenderingEngine.cpp

```diff
@@ -390,20 +390,28 @@ HRESULT CDX9RenderingEngine::RenderVideoDrawPath(IDirect3DSurface9* pRenderTarge
             start = stop;    // reset after 10 min (ps float has its limits in both range and accuracy)
         }
 
+        LONGLONG lCurrentPosition = 0;
+        LONGLONG lDuration = 0;
+        CComPtr<IMediaStatusPresenter> pInfos;
+        if (SUCCEEDED(GetMediaStatusPresenter(&pInfos))) {
+            pInfos->GetMediaTimes(&lCurrentPosition, &lDuration);
+        }
+        const float clTickToSeconds = 1.0f / 10000000.0f;
+
 #if 1
         D3DSURFACE_DESC desc;
         m_pVideoTexture[m_nCurSurface]->GetLevelDesc(0, &desc);
 
         float fConstData[][4] = {
-            {(float)desc.Width, (float)desc.Height, (float)(counter++), (float)diff / CLOCKS_PER_SEC},
-            {1.0f / desc.Width, 1.0f / desc.Height, 0, 0},
+            { (float)desc.Width, (float)desc.Height, (float)(counter++), (float)diff / CLOCKS_PER_SEC },

```

Comment:
> We are working on adding an external .json to set custom shader attributes (textures & others), not sure if it should be pushed too, it'll be on github anyway.

If you can wrap it into an interface that could be used by external video renderers such as madVR, then by all means yes.